### ISSUE TITLE
Replace forcing for gx1

### ISSUE THREAD

**Issue by eclare108213** [CICE-Consortium/CICE#249]
Prepare a new forcing data set for gx1, to replace the 'LYq' option that currently includes a mixture of COREII and other forcing with calculated radiation fields.   See #237.

**Comment by eclare108213** [comment]
The JRA-55 forcing has been implemented.   What remains is to retire the CORE-II based forcing.  I recommend removing it entirely from the code (forcing routines would still be available via github) and from the standard forcing data download (keep a link to it, but don't put it in the mass-download file).  This transition also includes QC tests, to make sure they still work as advertised.  @rallard77, didn't you document runs showing differences between old and new forcing?  If so, please link them here.  If not, that needs to be added to this task.  Thx

**Comment by eclare108213** [comment]
Would it be possible to make new gx3 forcing from JRA-55, too?

**Comment by dabail10** [comment]
And maybe tx1 as well?

**Comment by rallard77** [comment]
[jra55-example1.pptx](https://github.com/CICE-Consortium/CICE/files/3780390/jra55-example1.pptx)
Here just a few slides showing CORE II (slide 1) and JRA55 (slides 2&3). I have a lot more I can share (36 slides) if interested. It should be easy to also add gx3 and tx1.

**Comment by duvivier** [comment]
@rallard77 @eclare108213 
I have posted the output here:
https://github.com/CICE-Consortium/CICE/wiki/CICE-6.0.2-Sample-Output
In there I link to the 6.0.0 output since that is all still relevant for other configurations. 

I also updated the link in the release table for 6.0.2:
https://github.com/CICE-Consortium/CICE/wiki/CICE-Version-Index

You should be able to edit these wiki pages yourself if you would like, or you can let me know changes to make. Sorry I hadn't finished it before the release. I'd intended to do it that day, but got behind. 

For action items, @eclare108213 I want to clarify:
You want us to remove the COREII forcing entirely from the forcing data or to have a small separate file? This will also require changes to the wiki page on input data: https://github.com/CICE-Consortium/CICE/wiki/CICE-Input-Data. I don't want to modify the forcing files etc. until we have the code changes ready.

**Comment by eclare108213** [comment]
What I'd like is to move users off of COREII completely, in favor of JRA55, but I don't think we should remove the COREII data completely from the ftp site, for backwards compatibility.  I'm not sure what the best way to do this is.  We can certainly stop serving COREII unless it is specifically requested (i.e. don't package it as part of the standard forcing data).  And I agree, don't change the ftp until the code and script changes are ready.

**Comment by duvivier** [comment]
@eclare108213 Great, thanks for clarifying. I agree it makes sense to just support one in the master version of the code and maybe somewhere at the bottom of the wiki link to the COREII tarball that just includes that data and nothing else. I just looked - the CORE data are 27G (uncompressed) compared to the 47G for JRA55, so it'll be nice to get these out of the "all" tarball, but it's not too crazy big. 

Have you assigned anyone to the removal through the code and tests?

**Comment by eclare108213** [comment]
At the moment, it's assigned to @duvivier and @rallard77   :)
Others can help, of course

**Comment by duvivier** [comment]
@eclare108213 What's the status on this? I know we've updated the physical forcing files and released most those, but we still need to change defaults in the code to use JRA55 instead of COREII? Is that right? Should we add COREII forcing to our deprecated features list with a date TBD?

**Comment by eclare108213** [comment]
The forcing is already scheduled on the deprecated code wiki:
https://github.com/CICE-Consortium/CICE/wiki/DEPRECATED-CODE-FEATURES
There are a number of things that need to happen first, though, so the dates might change.

**Comment by duvivier** [comment]
Well, I feel like an idiot now. Ignore previous comment.

**Comment by eclare108213** [comment]
No problem!  We're both working on a scheduled holiday...  that's the idiocy.

**Comment by eclare108213** [comment]
An agenda and cursory notes from this morning's meeting to discuss the next steps in the JRA-55 transition are 
[here](https://docs.google.com/document/d/1An1-ZqlU_lyf84_qkJmyErUKnUDTy8aukvqyOtWCAWE/edit?usp=sharing) with a list of [action items](https://docs.google.com/document/d/1An1-ZqlU_lyf84_qkJmyErUKnUDTy8aukvqyOtWCAWE/edit#bookmark=id.tlj77rsfp3f7).

One thing we didn't discuss is what configuration to use for the spinup and "standard" test runs for which we will provide sample output.  The current [JRA-55 plots](https://github.com/CICE-Consortium/CICE/wiki/CICE-6.0.2-Sample-Output#cice-version-602---gx1-JRA55-forcing) used default gx1 namelist settings.  Does anyone out there have a preferred CICE parameter/parameterization configuration for JRA-55 forcing, or should we stick with the defaults?  Since the Consortium is providing this data for testing purposes only, I'm okay with using the defaults, although I'd like to at least review them to make sure they are physically sensible before doing the spinup.  I guess we'd know after the spinup if they are way out of kilter.

Attendees:  @eclare108213 @dabail10 @rallard77 @daveh150 

Related issues:
Issue [#481](https://github.com/CICE-Consortium/CICE/issues/481)  JRA forcing initialization/spinup
Issue [#186](https://github.com/CICE-Consortium/CICE/issues/186)  one-degree tripole support

**Comment by dabail10** [comment]
I personally think we should just use the default namelist configuration.

**Comment by eclare108213** [comment]
@dabail10 @rallard77 @daveh150 @apcraig @rgrumbine 

Per yesterday's meeting notes:
>We will combine efforts to convert to JRA forcing, incorporate NOAA metrics, and improve our testing coverage as discussed in last month’s meeting.
Action:  Elizabeth will schedule a series of meetings to coordinate this effort.

I propose that we use the **_Thursday 2 pm MT time slot for the next month_** (not Thanksgiving) to work through some aspects of this effort.  **_Would that work for all of you?_**  I’m hoping that we could do this in a few joint meetings with additional ‘homework’ in between.  Here’s what I’m thinking - open to other ideas - and maybe some of this is already done.  What's missing?

homework prior to Nov 19:

- run standard tests for current and new forcing data on as many platforms as possible, and create timeseries plots (each person in the meeting should be able to look at their own tests)
- review default namelist configuration

Nov 19

- compare basic output from JRA-55 and bulk forcing for the standard tests
- review code coverage to see what major things are missing
- decide whether to change any namelist values for 'production' runs
- discuss Bob’s metrics and decide which ones we can use to compare the gx1 production runs

homework Nov 20-Dec 3

- add tests or issues for specific code coverage needs
- test new scripts for extending JRA forcing years, create extended dataset
- run equivalent gx1 production cases for JRA and CORE forcing (how many years? who will be responsible for this?)

Dec 3

- review gx1 results
- discuss how to implement/apply Bob’s metrics

homework Dec 4-10

- implement/apply Bob’s metrics as discussed
- create a wiki page comparing the gx1 production runs

Dec 10

- share results with the full team
- decide when to deprecate old forcing

homework Dec 10-17+

- update CICE sample output https://github.com/CICE-Consortium/CICE/wiki/CICE-6.0.2-Sample-Output
- release code (or this can wait for a release next year)

December 17

- meet again if needed

**Comment by rallard77** [comment]
Elizabeth,

This sounds reasonable. I should be able to make the Nov 19 meeting.

Rick

**Comment by dabail10** [comment]
I can make the Nov 19 meeting as well.

**Comment by rgrumbine** [comment]
I was on vacation, but will be catching up. The Dec 4-10 homework should be good. Look for emails.

**Comment by apcraig** [comment]
It looks like use_leap_years is true for the JRA55 forcing configurations.  I don't have a lot of confidence in that option, but maybe others do.  The new time manager will carefully test leap and no leap capabilities but that's not ready yet.  We should make sure the calendar, forcing, and model are behaving properly over time and especially across leap days when we do longer runs using the JRA55 forcing.  Maybe that has already been done and checked, but I just want to call it out if it hasn't.

**Comment by phil-blain** [comment]
@apcraig regarding leap years: we have #343 already opened (just a reminder in case you did not recall that one)

**Comment by apcraig** [comment]
#343 is partly why I don't have much confidence in the leap calendar.  Two things have to work.  The CICE time manager has to handle the leap calendar properly.  That means dealing with changing number of days per year, timestep counts, restart dates, and all the rest.  The second is that reading the forcing has to work.  Does the CICE calendar have to match the forcing calendar?  Can the CICE calendar be set different from the forcing calendar?  I have many open questions, but was not involved in debugging the leap year problems in #343 and why I'm unclear on the exact status.  It's not clear everything is sorted out in #343, but maybe it is.

**Comment by phil-blain** [comment]
No, everything is not sorted out - I debugged a little bit an came to the conclusion that it was better if this issue was investigated thoroughly when the time manager is overhauled.

**Comment by apcraig** [comment]
From #533, 

To-do:

- [ ] Change default emissivity to 0.985
- [ ] Change default nblyr to 1

Notes from global/ (these are hemispheric values produced by print_global=T):

We should turn on the brine tracer for these tests. When it's off, maybe we shouldn't print this diagnostic.

- [ ] Should there be rain water in the Arctic in Jan/Feb/Mar?
- [x] The arwt tot mass, energy, salt diagnostics are wrong.

Notes from point/ (these are values for particular lat/lon points produced by print_point=T):

- [x] Arctic SST is wrong: large spikes, 2 lines, y axis values are wrong magnitude. This might just be a diagnostic issue.
- [ ] Why is the Antarctic shortwave radiation sum = 0 in mid summer?
- [ ]  Why is the snow change always 0?

The following issue may be more about the physical representation than diagnostic calculations. From the point plots:

- [ ] Why does gx1 Antarctic heat used go to an approximately constant (barely decreasing), nonzero value in mid-Feb? The area fraction ~0; bottom, top & lateral melt stops but thickness continues to decrease, and sensible, latent heat, outward longwave flux are all nonzero. The gx3 behavior is the opposite - Antarctic heat used is larger after mid-Feb, and the various terms remain nonzero. It could just be that the grid cells we are looking at are experiencing different conditions and hit the ice/no-ice threshold differently, but they should be fairly close to each other (lat/lon) so I wouldn't expect the conditions to be that different.

**Comment by apcraig** [comment]
See also #533

**Comment by eclare108213** [comment]
I'm struggling to download the gx1 and tx1 forcing - the files are too big.  Since most of our testing uses 1 year or less of the data, maybe we should break up the big JRA forcing files into 1-year chunks?  The other 4 years could also be put on zenodo (the full 5 year files would remain there for people who can get them), and then we can also offer the scripts to make the data files, once they are ready, if downloading them is a problem.  Thoughts?

**Comment by rgrumbine** [comment]
When I was downloading the files, they were already in 1 year chunks, which was manageable for me.  I do favor making things accessible to people with less bandwidth or computing power. Annual files aren't that much smaller than 5 years. So I'll suggest a larger number of monthly files. By way of my experiences distributing data for marine branch products, monthly works better for people than quarterly.

**Comment by dabail10** [comment]
Where are you downloading from Elizabeth? It does look like we have all five years in a single tar file for the CORE forcing. What is strange is that the JRA file (18GB) is much smaller than the CORE file (38GB). I think the JRA is only a single year. I'll do some digging.

**Comment by eclare108213** [comment]
I was downloading from zenodo, using the links on the github input data pages.  I wasn't downloading the CORE forcing, since I already have that.  I was able to get the gx3 JRA data onto my laptop (wireless), and all of it onto badger at LANL (using an actual internet cable), so I think I'm good to go for now.  But it was frustrating - the big JRA downloads kept hanging. I'm wondering if we need to make it a little more user-friendly, but if users aren't complaining, maybe we don't need to worry about it.  Internet here is notoriously bad.  I think the JRA data is 5 years per file.

**Comment by duvivier** [comment]
I believe we were trying to strike a balance between having files that were too big vs. having to download a bunch of files and possibly losing track of what you'd downloaded.

**Comment by dabail10** [comment]
Agreed. There was also the issue of Travis downloading the forcing.

**Comment by daveh150** [comment]
On Zenodo it looks like the gx3 data is 3.9GB, and the gx1 data is 39.8 GB. The gx1 in particular seems very large for one file. gx3 I would consider large for downloading to my work laptop, but I don’t do any coding on the laptop. 

 

We could break the files up into one year segments. For gx1 that is about 7.5 GB per year. 5 files, one per year, seems reasonable. Monthly files would result in 60 (12/year for 5 years). That seems like too much.

**Comment by apcraig** [comment]
travis (moving to github workflows) can be quickly changed to download any number of files.  it's hardwired into the script and doesn't matter much whether it's 1 file or 100 in the end.  For the CI testing, the main thing that might be useful would be to create reduced dataset if possible to support the CI testing.  But for now, the CI testing is just downloading some gx3 datasets (no gx1 testing in CI), so it's working pretty well.

**Comment by dabail10** [comment]
I forgot that it was only gx3 in Travis testing. I will go ahead and add single year files. This will give users flexibility.

**Comment by apcraig** [comment]
I think many of the issues here have been addressed, including questions about the calendar and JRA55 forcing.  I think those problems have been sorted out.  There are still some outstanding science questions, see checklist above.  

The main point seems to be that we have generated gx3, gx1, and tx1 JRA55 forcing datasets and have now carried out longer runs.  We're still fine tuning those and looking at the science.  Should we close and start a new issue about JRA55 results?
